### ISSUE TITLE
function: Illegal function name '-

### ISSUE THREAD

**Issue by emilstahl** [oh-my-fish/oh-my-fish#487]
```
fish, version 2.5.0

esp@esp ~> curl -L http://get.oh-my.fish | fish
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   154  100   154    0     0    248      0 --:--:-- --:--:-- --:--:--   248
100   140  100   140    0     0    133      0  0:00:01  0:00:01 --:--:--  136k
100 16533  100 16533    0     0  12622      0  0:00:01  0:00:01 --:--:-- 12622
Existing installation detected at /Users/esp/.local/share/omf
Would you like to remove the existing installation? (y/N): y
This will uninstall Oh My Fish and all plugins and themes from
/Users/esp/.local/share/omf.
Are you sure you want to continue? (y/N): y
Uninstalling from /Users/esp/.local/share/omf...
Uninstall complete
Installing Oh My Fish to /Users/esp/.local/share/omf...
Using release channel "stable".
Cloning master from https://github.com/oh-my-fish/oh-my-fish.git...
Writing bootstrap to /Users/esp/.config/fish/conf.d/omf.fish...
Setting up Oh My Fish configuration...
function: Illegal function name '-'
- (line 1): function - --wraps cd --description 'alias -=cd -';  cd - $argv; end
            ^
from sourcing file -
	called on line 68 of file /usr/local/Cellar/fish/2.5.0/share/fish/functions/alias.fish

in function 'alias'
	called on line 3 of file ~/.config/fish/functions/aliases.fish
	with parameter list '-=cd -'

from sourcing file ~/.config/fish/functions/aliases.fish
	called on line 221 of file /usr/local/Cellar/fish/2.5.0/share/fish/config.fish

in function '.'
	called on line 1 of file ~/.config/omf/init.fish

from sourcing file ~/.config/omf/init.fish
	called on line 74 of file ~/.local/share/omf/lib/require.fish

in function 'require'
	called on line 44 of file ~/.local/share/omf/init.fish
	with parameter list '--no-bundle --path /Users/esp/.config/omf'

from sourcing file ~/.local/share/omf/init.fish
	called on line 7 of file ~/.config/fish/conf.d/omf.fish

from sourcing file ~/.config/fish/conf.d/omf.fish
	called on line 241 of file /usr/local/Cellar/fish/2.5.0/share/fish/config.fish

from sourcing file /usr/local/Cellar/fish/2.5.0/share/fish/config.fish
	called during startup

Installing package foreign-env
✔ package foreign-env successfully installed.
Installing package nvm
✔ package nvm successfully installed.
Installing package rbenv
✔ package rbenv successfully installed.
Installing package sublime
✔ package sublime successfully installed.
Installing theme default
✔ theme default successfully installed.
Installation successful!
function: Illegal function name '-'
- (line 1): function - --wraps cd --description 'alias -=cd -';  cd - $argv; end
            ^
from sourcing file -
	called on line 68 of file /usr/local/Cellar/fish/2.5.0/share/fish/functions/alias.fish

in function 'alias'
	called on line 3 of file ~/.config/fish/functions/aliases.fish
	with parameter list '-=cd -'

from sourcing file ~/.config/fish/functions/aliases.fish
	called on line 221 of file /usr/local/Cellar/fish/2.5.0/share/fish/config.fish

in function '.'
	called on line 1 of file ~/.config/omf/init.fish

from sourcing file ~/.config/omf/init.fish
	called on line 74 of file ~/.local/share/omf/lib/require.fish

in function 'require'
	called on line 44 of file ~/.local/share/omf/init.fish
	with parameter list '--no-bundle --path /Users/esp/.config/omf'

from sourcing file ~/.local/share/omf/init.fish
	called on line 7 of file ~/.config/fish/conf.d/omf.fish

from sourcing file ~/.config/fish/conf.d/omf.fish
	called on line 241 of file /usr/local/Cellar/fish/2.5.0/share/fish/config.fish

from sourcing file /usr/local/Cellar/fish/2.5.0/share/fish/config.fish
	called during startup
```

**Comment by oranja** [comment]
Hi @emilstahl,

Do you mind posting the contents of:
(Make sure you remove any sensitive information though)
~/.config/omf/init.fish
~/.config/fish/functions/aliases.fish

Did you set `-` as an alias for `cd`? Do you use it?

**Comment by emilstahl** [comment]
@oranja Yes, I had ``- as an alias for `cd`. However I don't use it, so I just removed it. It works now.
Would you still like to see the contents?

**Comment by oranja** [comment]
No, it's not necessary, now that my theory was confirmed.
I'm glad it's sorted out.

For the protocol, here's the related change in fish 2.5.0: https://github.com/fish-shell/fish-shell/issues/2827
`alias` is a simply a *fish* function that creates a function with the name of the alias. In the case of `alias - 'cd'`, it would try to create `function -`, but *fish* function names are not allowed to start with `-`.
It seemed like everything is OK, because of a bug in previous *fish* versions, but it was fixed in 2.5.0, and now the failure of setting this alias up is not silenced.

Somewhere there's a definition of `alias - 'cd'`. It was never really allowed 
The missing link is that all `alias` does is create a simple function with the name of the alias. In this case, `function -`, and `-` is not a valid function name.

**Comment by emilstahl** [comment]
@oranja I see. Thanks for the explanation! 👍
